pkg/secretutils/label.go

```diff
@@ -0,0 +1,30 @@
+package secretutils
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+)
+
+const (
+	LabelEnvironmentName  = "environment.metal3.io"
+	LabelEnvironmentValue = "baremetal"
+)
+
+// CacheLabelSelector returns a Label Selector that filters Objects that have
+// been labelled as part of the baremetal environment.
+func CacheLabelSelector() labels.Selector {
+	return labels.SelectorFromSet(labels.Set{
+		LabelEnvironmentName: LabelEnvironmentValue,
+	})
+}
+
+// CacheSecretSelectors returns a SelectorsByObject that filters Secrets so
+// that only those labelled as part of the baremetal environment get cached.
+func CacheSecretSelectors() cache.SelectorsByObject {
+	return cache.SelectorsByObject{
+		&corev1.Secret{}: {
+			Label: CacheLabelSelector(),
+		},
+	}
+}
```

Comment:
This approach doesn't look to scale well in case in future there could be the need to specify an additional cache selector for another kind of resource different from a secret (and maybe managed by another manager). One approach could be to transform this method so that it could receive a `cache.SelectorsByObject` and fill it for just the secret type, leaving the possibility to other contributors to set eventually other types.